src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,261 @@ public static class NUnitTestExtractorApp
+                    continue;
                 }
-                return false;
-            }
-            throw new ArgumentNullException("excludeInfo", "no excludeInfo specified");
-        }
 
-        /// <summary>
-        /// Depending on number of args within the TestCase, function will return properly formatted testcase
-        /// </summary>
-        /// <param name="declaringType">declaring type of testcase method(namespace.class)</param>
-        /// <param name="name">name of testcase</param>
-        /// <param name="testCaseArgs">the object array containing the testcase's arguments</param>
-        /// <param name="data">the string which is appended to</param>
-        /// <returns>the properly formatted TestCase string</returns>
-        public static string FormattedTestCase(string declaringType, string name, object[] testCaseArgs, string data)
-        {
+                // search the category attributes for the function
+                var categories = attributes.Where(x => x.AttributeType.Equals(typeof(CategoryAttribute)));
 
-            if (testCaseArgs != null && testCaseArgs.Length > 0)
-            {
-                data = declaringType + "." + name + "(";
-                data = AppendArgumentsForTestCasesToString(testCaseArgs, data);
-                data += ")";
-            }
-            else if (testCaseArgs != null && testCaseArgs.Length == 0)
-            {
-                data = declaringType + "." + name + "()";
+                // match the "Exclude" category from the function category attibute list
+                // if the function has matched exclude category, we will skip this function
+                if (categories.Any(x => x.ConstructorArguments.Any(y => y.Value.Equals(_options.ExcludeInfo))))
+                {
+                    continue;
+                }
+
+                // match the "Include" category from the function category attribute list
+                // if the function has matched include category, we will count this function as a valid test
+                if (string.IsNullOrWhiteSpace(_options.IncludeInfo)
+                    || categories.Any(x => x.ConstructorArguments.Any(y => y.Value.Equals(_options.IncludeInfo))))
+                {
+                    tests.Add(test.DeclaringType.GetTypeInfo().FullName + "." + test.Name);
+                }
             }
 
-            return data;
+            return tests;
         }
 
         /// <summary>
-        /// Parses string into proper Level
+        /// Create the output file if it doesn't exist, or append it if it exists
         /// </summary>
-        /// <param name="levelToParse">The string to parse</param>
-        /// <returns>The Level from parsing the string</returns>
-        public static Level ParseLevel(string levelToParse)
+        /// <param name="outputInfo">The raw information need to be re-construct to our expected format.</param>
+        private static void OutputInfo(HashSet<string> outputInfoList)
```

Comment:
[Should] The parameter name here is outputInfoList but in the `<param>` tag it's outputInfo.  They should match.